pkg/services/ngalert/api/api_ruler.go

```diff
@@ -307,23 +329,24 @@ func (srv RulerSrv) updateAlertRulesInGroup(c *models.ReqContext, namespace *mod
 		return ErrResp(http.StatusInternalServerError, err, "failed to update rule group")
 	}
 
-	// TODO uncomment when rules that are not changed will be filter out from the upsert list.
-	// for _, rule := range changes.Upsert {
-	// 	if rule.Existing != nil {
-	// 		srv.scheduleService.UpdateAlertRule(ngmodels.AlertRuleKey{
-	// 			OrgID: c.SignedInUser.OrgId,
-	// 			UID:   rule.Existing.UID,
-	// 		})
-	// 	}
-	// }
+	for _, rule := range groupChanges.Update {
+		srv.scheduleService.UpdateAlertRule(ngmodels.AlertRuleKey{
+			OrgID: c.SignedInUser.OrgId,
```

Comment:
Do we need this if `groupChanges.Update` is also upserted via `UpsertAlertRules`?

Reply:
The reason we do this here is to avoid confusion on the user side because we expire alert instances immediately, not after 10s. Without it update\deletion of a rule will not be immediately reflected in Alert instance tab.